### DISCUSSION TITLE
Decrementing build number on build failure?

### DISCUSSION THREAD

**Discussion by swrobel** [fastlane/fastlane#19367] [Q&A]
Is there a way to recognize that a build failed and decrement the build number? This was raised by @mlostekk previously in https://github.com/fastlane/fastlane/issues/5312 but there was no reply

**Comment by crazymanish** [comment]
we don't need **decrement_version_number** instead i will suggest below:
- Use increment_version_number and git-commit for bumped version_number
- If case of failure, **revert** bumped version_number git-commit

i hope this will help!

**Reply by swrobel** [reply]
How do I actually trap the failure & take action?

**Reply by crazymanish** [reply]
Using Ruby exception-handling block
https://scoutapm.com/blog/exception-handling-in-ruby

example:
```ruby
lane :test do
      begin
        increment_version_number
        # some code that can fail
      rescue => exception
        # handle exception in this block
        # revert/retry something or whatever you want to do based on your workflow
      end
 end
